setup/src/installer.ts

```diff
@@ -196,7 +199,10 @@ async function choco(tool: Tool, version: string): Promise<void> {
   ]);
   console.log('::SetupHaskellStopCommands::'); // Re-enable command execution
   // Add GHC to path automatically because it does not add until the end of the step and we check the path.
-  if (tool == 'ghc') core.addPath(getChocoPath(tool, version));
+
+  const chocoPath = await getChocoPath(tool, version);
+
+  if (tool == 'ghc') core.addPath(chocoPath || 'undefined');
```

Comment:
This doesn't make a lot of sense to me. If I'm reading this right, if chocoPath is null, we add 'undefined' to the path?

I would expect the entire action to fail if chocolatey wasn't installed and the right tools weren't already setup correctly. There's no other way to install GHC on windows that this action knows about so it can't meaningfully progress past the initial no-op "check if tool is already on the path"

Reply:
I've removed the `'undefined'` here as well.